packages/core/src/oidc/init.ts

```diff
@@ -145,8 +142,22 @@ export default function initOidc(
 
           const { accessTokenTtl: accessTokenTTL } = resourceServer;
 
-          const scopes = await findResourceScopes(queries, libraries, ctx, indicator);
-          const { client } = ctx.oidc;
+          const { client, params } = ctx.oidc;
+          /**
+           * In consent or code excange flow, the organization_id is undefined,
+           * and all the scopes inherited from the all organization roles will be granted.
+           * In the flow of granting token for organization with api resource,
+           * this value is set to the organization id,
+           * and will then narrow down the scopes to the specific organization.
+           */
```

Comment:
nit: feels like such a comment should be placed with in the `findResourceScopes` function definition.  e.g Put it as part of the method's jsdoc comment. Indicate the usage of `param.organizationId`

Reply:
We also have comments for function `findResourceScopes`.